service/src/test/java/uk/gov/hmcts/reform/fpl/service/DraftCMOServiceTest.java

```diff
@@ -0,0 +1,117 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {JacksonAutoConfiguration.class, DateFormatterService.class, DraftCMOService.class})
+public class DraftCMOServiceTest {
+
+    @Autowired
+    private DraftCMOService draftCMOService;
+
+    private final LocalDate date = LocalDate.now();
+
+    @Test
+    void shouldReturnHearingDateDynamicListWhenCaseDetailsHasHearingDate() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of(
+                "hearingDetails", createHearingBookings(date))
+            )
+            .build();
+
+        DynamicList hearingList = draftCMOService.getHearingDatesDynamic(caseDetails);
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
```

Comment:
can you do hearingList.getListItems().contains as your assertion? You can chain contains

Reply:
done